back/lib/tasks/single_use/20240910_migrate_initiatives_to_proposals.rake

```diff
@@ -0,0 +1,811 @@
+
+    {
+      'type' => { 'resolvedName' => 'Highlight' },
+      'nodes' => [],
+      'props' => {
+        'title' => rake_20240910_migrate_homepage_title_multiloc,
+        'description' => rake_20240910_migrate_homepage_description_multiloc,
+        'primaryButtonLink' => "/projects/#{project.slug}",
+        'primaryButtonText' => rake_20240910_migrate_homepage_explore_multiloc,
+        'secondaryButtonLink' => '',
+        'secondaryButtonText' => {}
+      },
+      'custom' => {
+        'title' => {
+          'id' => 'app.containers.admin.ContentBuilder.homepage.highlight.highlightTitle',
+          'defaultMessage' => 'Highlight'
+        },
+        'noPointerEvents' => true
+      },
+      'hidden' => false,
+      'parent' => 'ROOT',
+      'isCanvas' => false,
+      'displayName' => 'Highlight',
+      'linkedNodes' => {}
+    }
+  end
+  if !homepage.save
+    reporter.add_error(
+      homepage.errors.details,
+      context: { tenant: AppConfiguration.instance.host, homepage: homepage.id }
+    )
+  end
+end
+
+def rake_20240910_replace_navbaritem(project, reporter)
+  item = NavBarItem.find_by(code: 'proposals')
+  return if !item
+
+  item.assign_attributes(
+    code: 'custom',
+    project_id: project.id
+  )
+  item.title_multiloc = MultilocService.new.i18n_to_multiloc('nav_bar_items.proposals.title', locales: CL2_SUPPORTED_LOCALES) if item.title_multiloc.blank?
+  if !item.save
+    reporter.add_error(
+      item.errors.details,
+      context: { tenant: AppConfiguration.instance.host, nav_bar_item: item.id }
+    )
+  end
+end
+
+def rake_20240910_migrate_input_statuses(reporter)
+  IdeaStatus.where.not(code: 'custom').each do |status|
+    if status.participation_method == 'proposals'
+      status.title_multiloc = MultilocService.new.i18n_to_multiloc("idea_statuses.#{status.code}", locales: CL2_SUPPORTED_LOCALES)
+      # TODO: replace colours of proposal statuses
+      # TODO: identify proposal statuses with custom copy
+    end
+    status.description_multiloc = MultilocService.new.i18n_to_multiloc("idea_statuses.#{status.code}_description", locales: CL2_SUPPORTED_LOCALES)
+    if !status.save
+      reporter.add_error(
+        status.errors.details,
+        context: { tenant: AppConfiguration.instance.host, input_status: status.id }
+      )
+    end
+  end
+end
+
+def rake_20240910_proposal_attributes(initiative, project)
+  proposals_phase = project.phases.first
+  {
+    slug: SlugService.new.generate_slug(Idea.new, initiative.slug),
+    title_multiloc: initiative.title_multiloc,
+    body_multiloc: initiative.body_multiloc,
+    author_id: initiative.author_id,
+    created_at: initiative.created_at,
+    updated_at: initiative.updated_at,
+    location_point_geojson: initiative.location_point_geojson,
+    location_description: initiative.location_description,
+    assignee_id: initiative.assignee_id,
+    assigned_at: initiative.assigned_at,
+    author_hash: initiative.author_hash,
```

Comment:
Yes that is correct. If the author_hash is migrated as it is, then if the same person subsequently contributes anonymously they will appear as a different author. However, I've just taken a look at metabase and this is a non-issue given that there are only 12 initiatives in the entire system that have been posted anonymously, and this author_hash is only used in those circumstances

Reply:
Good point, will check with James. I'm not sure if an alternative solution like regenerating the author hashes would be an option, as this would probably make the script a lot slower.

Reply:
It's fine to leave as it is